? wxgtkd-2.3-config
? lexer.c
? parser.c
? new_button.patch
? contrib/config.log
? contrib/config.status
? contrib/samples/Makefile
? contrib/samples/applet/Makefile
? contrib/samples/canvas/Makefile
? contrib/samples/canvas/simple/Makefile
? contrib/samples/canvas/test/Makefile
? contrib/samples/fl/Makefile
? contrib/samples/fl/fl_demo1/Makefile
? contrib/samples/fl/fl_demo2/Makefile
? contrib/samples/fl/fl_sample1/Makefile
? contrib/samples/fl/fl_sample2/Makefile
? contrib/samples/fl/fl_sample3/Makefile
? contrib/samples/gizmos/Makefile
? contrib/samples/gizmos/dynsash/Makefile
? contrib/samples/gizmos/dynsash_switch/Makefile
? contrib/samples/gizmos/editlbox/Makefile
? contrib/samples/gizmos/multicell/Makefile
? contrib/samples/gizmos/splittree/Makefile
? contrib/samples/mmedia/Makefile
? contrib/samples/ogl/Makefile
? contrib/samples/ogl/ogledit/Makefile
? contrib/samples/ogl/studio/Makefile
? contrib/samples/plot/Makefile
? contrib/samples/stc/Makefile
? contrib/samples/svg/Makefile
? contrib/samples/xrc/Makefile
? contrib/src/Makefile
? contrib/src/applet/Makefile
? contrib/src/canvas/Makefile
? contrib/src/fl/Makefile
? contrib/src/gizmos/Makefile
? contrib/src/mmedia/Makefile
? contrib/src/net/Makefile
? contrib/src/ogl/Makefile
? contrib/src/plot/Makefile
? contrib/src/stc/Makefile
? contrib/src/svg/Makefile
? contrib/src/xrc/Makefile
? contrib/utils/Makefile
? contrib/utils/convertrc/Makefile
? contrib/utils/wxrc/Makefile
? contrib/utils/wxrcedit/Makefile
? demos/config.log
? demos/config.status
? demos/Makefile
? demos/bombs/Makefile
? demos/dbbrowse/Makefile
? demos/forty/Makefile
? demos/fractal/Makefile
? demos/life/Makefile
? demos/poem/Makefile
? lib/wx
? lib/libwx_gtkd-2.3.so.2.0.0
? lib/libwx_gtkd_gl-2.3.so.2.0.0
? samples/config.log
? samples/config.status
? samples/Makefile
? samples/artprov/Makefile
? samples/calendar/Makefile
? samples/caret/Makefile
? samples/checklst/Makefile
? samples/config/Makefile
? samples/console/Makefile
? samples/controls/Makefile
? samples/db/Makefile
? samples/dialogs/Makefile
? samples/dialup/Makefile
? samples/dnd/Makefile
? samples/docview/Makefile
? samples/docvwmdi/Makefile
? samples/dragimag/Makefile
? samples/drawing/Makefile
? samples/dynamic/Makefile
? samples/erase/Makefile
? samples/event/Makefile
? samples/exec/Makefile
? samples/font/Makefile
? samples/grid/Makefile
? samples/help/Makefile
? samples/html/Makefile
? samples/html/about/Makefile
? samples/html/help/Makefile
? samples/html/helpview/Makefile
? samples/html/printing/Makefile
? samples/html/test/Makefile
? samples/html/virtual/Makefile
? samples/html/widget/Makefile
? samples/html/zip/Makefile
? samples/image/Makefile
? samples/internat/Makefile
? samples/ipc/Makefile
? samples/joytest/Makefile
? samples/keyboard/Makefile
? samples/layout/Makefile
? samples/listctrl/Makefile
? samples/mdi/Makefile
? samples/memcheck/Makefile
? samples/menu/Makefile
? samples/minifram/Makefile
? samples/minimal/Makefile
? samples/mobile/Makefile
? samples/mobile/styles/Makefile
? samples/mobile/wxedit/Makefile
? samples/newgrid/Makefile
? samples/notebook/Makefile
? samples/opengl/Makefile
? samples/opengl/cube/Makefile
? samples/opengl/isosurf/Makefile
? samples/opengl/penguin/Makefile
? samples/png/Makefile
? samples/printing/Makefile
? samples/proplist/Makefile
? samples/propsize/Makefile
? samples/resource/Makefile
? samples/richedit/Makefile
? samples/rotate/Makefile
? samples/sashtest/Makefile
? samples/scroll/Makefile
? samples/scrollsub/Makefile
? samples/sockets/Makefile
? samples/splitter/Makefile
? samples/statbar/Makefile
? samples/text/Makefile
? samples/thread/Makefile
? samples/toolbar/Makefile
? samples/treectrl/Makefile
? samples/treelay/Makefile
? samples/typetest/Makefile
? samples/validate/Makefile
? samples/widgets/Makefile
? samples/widgets/button.d
? samples/widgets/combobox.d
? samples/widgets/gauge.d
? samples/widgets/listbox.d
? samples/widgets/notebook.d
? samples/widgets/radiobox.d
? samples/widgets/slider.d
? samples/widgets/spinbtn.d
? samples/widgets/static.d
? samples/widgets/textctrl.d
? samples/widgets/widgets.d
? samples/widgets/widgets
? samples/wizard/Makefile
? utils/config.log
? utils/config.status
? utils/Makefile
? utils/HelpGen/Makefile
? utils/HelpGen/src/Makefile
? utils/dialoged/Makefile
? utils/emulator/Makefile
? utils/emulator/src/Makefile
? utils/helpview/Makefile
? utils/helpview/src/Makefile
? utils/hhp2cached/Makefile
? utils/makegen/Makefile
? utils/tex2rtf/Makefile
? utils/tex2rtf/src/Makefile
Index: include/wx/button.h
===================================================================
RCS file: /home/wxcvs/wxWindows/include/wx/button.h,v
retrieving revision 1.9
diff -b -u -2 -r1.9 button.h
--- include/wx/button.h	25 Aug 2001 14:52:25 -0000	1.9
+++ include/wx/button.h	16 Aug 2002 15:09:25 -0000
@@ -19,7 +19,9 @@
 // ----------------------------------------------------------------------------
 
-// all these flags are obsolete
+// These two flags are obsolete
 #define wxBU_NOAUTODRAW      0x0000
 #define wxBU_AUTODRAW        0x0004
+
+// These flags affect label alignment
 #define wxBU_LEFT            0x0040
 #define wxBU_TOP             0x0080
Index: samples/widgets/button.cpp
===================================================================
RCS file: /home/wxcvs/wxWindows/samples/widgets/button.cpp,v
retrieving revision 1.6
diff -b -u -2 -r1.6 button.cpp
--- samples/widgets/button.cpp	20 Apr 2002 23:30:23 -0000	1.6
+++ samples/widgets/button.cpp	16 Aug 2002 15:09:27 -0000
@@ -281,5 +281,5 @@
     {
         case ButtonHAlign_Left:
-            flags |= wxALIGN_LEFT;
+            flags |= wxBU_LEFT;
             break;
 
@@ -289,9 +289,8 @@
 
         case ButtonHAlign_Centre:
-            flags |= wxALIGN_CENTRE_HORIZONTAL;
             break;
 
         case ButtonHAlign_Right:
-            flags |= wxALIGN_RIGHT;
+            flags |= wxBU_RIGHT;
             break;
     }
@@ -300,5 +299,5 @@
     {
         case ButtonVAlign_Top:
-            flags |= wxALIGN_TOP;
+            flags |= wxBU_TOP;
             break;
 
@@ -312,5 +311,5 @@
 
         case ButtonVAlign_Bottom:
-            flags |= wxALIGN_BOTTOM;
+            flags |= wxBU_BOTTOM;
             break;
     }
Index: src/gtk/button.cpp
===================================================================
RCS file: /home/wxcvs/wxWindows/src/gtk/button.cpp,v
retrieving revision 1.50
diff -b -u -2 -r1.50 button.cpp
--- src/gtk/button.cpp	5 Aug 2002 17:59:19 -0000	1.50
+++ src/gtk/button.cpp	16 Aug 2002 15:09:27 -0000
@@ -108,4 +108,19 @@
     m_widget = gtk_button_new_with_label("");
 
+    float x_alignment = 0.5;
+    if (HasFlag(wxBU_LEFT))
+        x_alignment = 0.0;
+    else if (HasFlag(wxBU_RIGHT))
+        x_alignment = 1.0;
+
+    float y_alignment = 0.5;
+    if (HasFlag(wxBU_TOP))
+        y_alignment = 0.0;
+    else if (HasFlag(wxBU_BOTTOM))
+        y_alignment = 1.0;
+
+    gtk_misc_set_alignment (GTK_MISC (BUTTON_CHILD (m_widget)),
+			    x_alignment, y_alignment);
+
     SetLabel( label );
 

 	  	 
